rasa/nlu/classifiers/keyword_intent_classifier.py

```diff
@@ -1,32 +1,156 @@
-from typing import Any, Optional, Text
+import os
+import logging
+import typing
+import re
+from typing import Any, Dict, Optional, Text
 
+from rasa.nlu import utils
 from rasa.nlu.components import Component
 from rasa.nlu.training_data import Message
 
+logger = logging.getLogger(__name__)
+
+if typing.TYPE_CHECKING:
+    from rasa.nlu.config import RasaNLUModelConfig
+    from rasa.nlu.training_data import TrainingData
+    from rasa.nlu.model import Metadata
+    from rasa.nlu.training_data import Message
+
 
 class KeywordIntentClassifier(Component):
+    """Intent classifier using simple keyword matching.
+
+
+    The classifier takes a list of keywords and associated intents as an input.
+    A input sentence is checked for the keywords and the intent is returned.
+
+    """
 
     provides = ["intent"]
 
-    his = ["hello", "hi", "hey"]
+    defaults = {"case_sensitive": True}
+
+    def __init__(
+        self,
+        component_config: Optional[Dict[Text, Any]] = None,
+        intent_keyword_map: Optional[Dict] = None,
+    ):
+
+        super(KeywordIntentClassifier, self).__init__(component_config)
 
-    byes = ["bye", "goodbye"]
+        self.intent_keyword_map = intent_keyword_map or {}
+
+        self.re_case_flag = (
```

Comment:
let's use `self.case_senitive` here and change it to re thingy where it is used

Reply:
👍 moved it to a local var calculated in the methods that use it